src/test/java/redis/clients/jedis/ClusterPipeliningTest.java

```diff
@@ -0,0 +1,883 @@
+        Assert.assertEquals(expectedResponse, r9.get().get(0));
+        Assert.assertEquals(Long.valueOf(1), r10.get());
+        Assert.assertTrue(r11.get().size() == 1 && r11.get().contains("place1"));
+        Assert.assertTrue(r12.get().size() == 2 && r12.get().get(0).getMemberByString().equals("place2"));
+        Assert.assertTrue(r13.get().size() == 2 && r13.get().get(0).getMemberByString().equals("place2"));
+        Assert.assertTrue(r14.get().size() == 1 && r14.get().get(0).getMemberByString().equals("place2"));
+        Assert.assertTrue(r15.get().size() == 1 && r15.get().get(0).getMemberByString().equals("place2"));
+        Assert.assertEquals(Long.valueOf(1), r16.get());
+        Assert.assertTrue(r17.get().size() == 1 && r17.get().contains("place2"));
+    }
+
+    @Test
+    public void clusterPipelineHyperLogLog() {
+        ClusterConnectionProvider provider = new ClusterConnectionProvider(nodes, DEFAULT_CLIENT_CONFIG);
+        ClusterPipeline p = new ClusterPipeline(provider);
+
+        Response<Long> r1 = p.pfadd("{hll}_1", "foo", "bar", "zap", "a");
+        Response<Long> r2 = p.pfadd("{hll}_2", "foo", "bar", "zap");
+        Response<Long> r3 = p.pfcount("{hll}_1", "{hll}_2");
+        Response<String> r4 = p.pfmerge("{hll}3","{hll}_1", "{hll}_2");
+        Response<Long> r5 = p.pfcount("{hll}3");
+
+        p.sync();
+        Assert.assertEquals(Long.valueOf(1), r1.get());
+        Assert.assertEquals(Long.valueOf(1), r2.get());
+        Assert.assertEquals(Long.valueOf(4), r3.get());
+        Assert.assertEquals("OK", r4.get());
+        Assert.assertEquals(Long.valueOf(4), r5.get());
+    }
+
+    @Test
+    public void clusterPipelineStringsAndBits() {
+        List<Long> fieldRes = new ArrayList<>();
+        fieldRes.add(1L);
+        fieldRes.add(0L);
+
+        ClusterConnectionProvider provider = new ClusterConnectionProvider(nodes, DEFAULT_CLIENT_CONFIG);
+        ClusterPipeline p = new ClusterPipeline(provider);
+
+        Response<String> r1 = p.set("mykey{^}", "foobar"); // foobar = 66 6f 6f 62 61 72
+        p.set("myotherkey", "foo");
+        Response<String> r2 = p.substr("mykey{^}", 0,2);
+        Response<Long> r3 = p.strlen("mykey{^}");
+        Response<Long> r4 = p.bitcount("myotherkey");
+        Response<Long> r5 = p.bitcount("myotherkey", 1,1);
+        Response<Long> r6 = p.bitpos("mykey{^}", true);
+        Response<Long> r7 = p.bitpos("mykey{^}", false, new BitPosParams(1, 2));
+        Response<List<Long>> r8 = p.bitfield("mynewkey","INCRBY", "i5", "100", "1", "GET", "u4", "0");
+        Response<List<Long>> r9 = p.bitfieldReadonly("hello", "GET", "i8", "17");
+        p.set("myotherkey{^}", "abcdef");
+        Response<Long> r10 = p.bitop(BitOP.AND, "dest{^}", "mykey{^}", "myotherkey{^}");
+        Response<String> r11 = p.get("dest{^}");
+        Response<Boolean> r12 = p.setbit("myotherkey", 7, true);
+        Response<Boolean> r13 = p.getbit("myotherkey", 7);
```

Comment:
Could you please a better hashtag? E.g. `mykey`, as in `{mykey}`, `{mykey}other`, etc.